torch/csrc/utils/python_dispatch.cpp

```diff
@@ -738,6 +739,11 @@ void initDispatchBindings(PyObject* module) {
         include_set.has(c10::DispatchKey::FuncTorchDynamicLayerFrontMode) ||
         include_set.has(c10::DispatchKey::FuncTorchDynamicLayerBackMode));
   });
+
+  m.def("get_singleton_int", [](int64_t data) {
+    return c10::SymInt(
+        c10::SymNode(c10::make_intrusive<c10::SingletonSymNodeImpl>(data)));
```

Comment:
Hmm, I'm not sure I want to let people generate singletons by number, it may be safer to offer only a fresh singleton api

Reply:
See other comment https://github.com/pytorch/pytorch/pull/107089#discussion_r1294006232

Reply:
Made this a private API.